gnoi_target/gnoi_target.go

```diff
@@ -51,18 +56,27 @@ func serve() {
 	}
 }
 
-// notify can be called with the number of certs and ca certs installed. It will
+// notifyReset will be called when the factory reset service requires the server
+// to be restarted.
+func notifyReset() {
+	var err error
+	if gNOIServer, err = gnoi.NewServer(nil, nil, resetSettings); err != nil {
```

Comment:
things that you need to consider here:
1) you need to close the existing connections and server being handled in serve().
2) this is the place to introduce the delay, you can even send a log message.
3) you need to call notifyCerts at the end again to restart the server.

Reply:
I think doing number 3 will be enough for 1 since it gracefully shuts down the server